ReactJS.php

```diff
@@ -21,6 +21,12 @@ class ReactJS {
      * @var string
      */
     $component,
+
+    /**
+     * The Javascript scope in which React components are contained (defaults to global)
+     * @var string
+     */
+    $componentScope = 'global',
```

Comment:
Seems fine but this should default to `window`. `global` is not defined by default.

Or perhaps even better, we don't define a default and if it's not set, then we don't try to use it.

Also nit here and other comments: ""JavaScript", not "Javascript".